packages/vats/src/virtual-purse.js

```diff
@@ -48,15 +49,17 @@ export const makeVirtualPurseKitIKit = (
   });
 
   const RetainRedeemI = M.interface('RetainRedeem', {
-    retain: M.callWhen(PaymentShape).optional(amountShape).returns(amountShape),
+    retain: M.callWhen(PaymentShape)
+      .optional(AmountPatternShape)
+      .returns(amountShape),
     redeem: M.callWhen(amountShape).returns(PaymentShape),
   });
 
   const UtilsI = M.interface('Utils', {
     retain: getInterfaceGuardPayload(RetainRedeemI).methodGuards.retain,
     redeem: getInterfaceGuardPayload(RetainRedeemI).methodGuards.redeem,
     recoverableClaim: M.callWhen(M.await(PaymentShape))
-      .optional(amountShape)
+      .optional(AmountPatternShape)
```

Comment:
Should `AmountPatternShape` be a parameter to `makeVirtualPurseKitIKit` like the others?

Reply:
No. It should always be exactly an alias for `M.pattern()` since any pattern should be valid.

Resolving.